user_token.go

```diff
@@ -0,0 +1,136 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ UserTokens = (*userTokens)(nil)
+
+// UserTokens describes all the user token related methods that the
+// Terraform Cloud/Enterprise API supports.
+//
+// TFE API docs:
+// https://www.terraform.io/docs/enterprise/api/user-tokens.html
+type UserTokens interface {
+	// List all the tokens of the given user ID.
+	List(ctx context.Context, userID string) (*UserTokenList, error)
+
+	// Create a new user token
+	Create(ctx context.Context, userID string, options UserTokenCreateOptions) (*UserToken, error)
+
+	// Read a user token by its ID.
+	Read(ctx context.Context, tokenID string) (*UserToken, error)
+
+	// Delete a user token by its ID.
+	Delete(ctx context.Context, tokenID string) error
+}
+
+// userTokens implements UserTokens.
+type userTokens struct {
+	client *Client
+}
+
+// UserTokenList is a list of tokens for the given user ID.
+type UserTokenList struct {
+	*Pagination
+	Items []*UserToken
+}
+
+// UserToken represents a Terraform Enterprise user token.
+type UserToken struct {
+	ID          string    `jsonapi:"primary,authentication-tokens"`
+	CreatedAt   time.Time `jsonapi:"attr,created-at,iso8601"`
+	Description string    `jsonapi:"attr,description"`
+	LastUsedAt  time.Time `jsonapi:"attr,last-used-at,iso8601"`
+	Token       string    `jsonapi:"attr,token"`
+}
+
+// UserTokenCreateOptions the options for creating a user token.
+type UserTokenCreateOptions struct {
+	// Description of the token
+	Description string `jsonapi:"attr,description,omitempty"`
+}
+
+// Create a new user token, replacing any existing token.
```

Comment:
Probably just a copy/paste error; new user tokens do not replace any existing tokens, as you know.

Reply:
Yes, sloppy of me! Fixed in 38d6462